README.md

```diff
@@ -5,6 +5,7 @@
 [Java](https://www.java.com/en/) plugin for the [asdf](https://github.com/asdf-vm/asdf) version manager.
 
 ## Requirements
+- [bash v5.0](https://www.gnu.org/software/bash/)
```

Comment:
What was the requirement before?

Unfortunately macOS still comes with Bash v3.x (the last version licensed under GPLv2) and this would require people to update manually Bash, for example via Homebrew.

```
# /bin/bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)
Copyright (C) 2007 Free Software Foundation, Inc.
```

Reply:
Good question, this is something I was hoping to get some input on. We didn't have a known minimum bash version requirement. I did run into problems on line #113 using bash v3.x on a mac. I don't know exactly which version of bash adds support for the conditional being used on line #113. However, there are a number of features I'd like to be able to take advantage of that are not available in bash v3.x

Here are a few of my thoughts:
- Mac users already need to install jq and coreutils. I do wish Mac was distributed with a more modern version of bash, and I hate adding an additional dependency burden to them. However, by not requiring a lower minimum version of bash, we could be limiting our ability to make the plugin more robust and capable.
- Namerefs became available around bash v4.2 - but there were all kinds of bugs relating to circular reference checks that were worked out in higher versions, so I figure just skip all that and jump to v5.0. Appropriate use of namerefs will enable us to simplify and make the system more elegant.
- Requiring v5.0 may help us eliminate support problems in the future as people make contributions that might make use of features only available in higher versions than those installed by default
- Minimum versions of the other tools (like `jq`) may need to be specified

Reply:
Not a very good idea to force people to update the bash, I think. How about zsh? It will work well in Zsh?

Reply:
@donbeave asdf (and by extension all its plugins) are bash-only. 😉

Reply:
@halcyon I'm fine with requiring Bash 5.x for asdf-java. I just wanted it to be a conscious decision rather than an accidental requirement. 👍

Reply:
Maybe it's too early to switch to Bash 5, can it be Bash 4 for example? For macOS users, I think it's not a big deal, but probably for Linux users or users on Windows Subsystem for Linux (WSL) can be much more difficult to upgrade to Bash 5. But I can not be clear, because personally I'm using macOS and using brew it's easy to install a new bash, not sure it's the same easy for Windows/Linux users.

Reply:
The [ASDF plugin guide](https://github.com/asdf-vm/asdf/blob/master/docs/plugins-create.md) says that plugins should work with at least MacOS and Linux. I'm delighted when when an asdf plugin works on one of my other systems (like OpenBSD), but I'm not surprised when it doesn't. We don't officially support Windows at this time - see #26.

Reply:
Windows is not important, I mean WSL which is actually Ubuntu inside the Windows. Which almost the same as Ubuntu in VirtualBox or Docker.

Reply:
Yes, I understand - I'm aware of and have used WSL (and WSL2 for that matter), cygwin, mingw, etc. I'm sympathetic, but these platforms can be tricky to support at best.